src/components/visualizations/chart/legend/StaticLegend.tsx

```diff
@@ -68,16 +69,26 @@ export default class StaticLegend extends React.PureComponent<any, any> {
             );
         }
 
-        const { hasPaging, visibleItemsCount } = calculateStaticLegend(series.length, containerHeight);
+        const seriesCount = series.length;
+        const { hasPaging, visibleItemsCount } = calculateStaticLegend(seriesCount, containerHeight);
 
         const start = (page - 1) * visibleItemsCount;
         const end = Math.min(visibleItemsCount * page, series.length);
 
         const pagedSeries = series.slice(start, end);
 
+        // For Geo Pushpin with position left/right
+        // we set the height of series to number of actual displayed items
+        // so that, size legend will be visible
+        const isPushpinAndLeftRightPosition =
+            isPushpin(chartType) && (position === LEFT || position === RIGHT);
+        const seriesHeight =
+            (isPushpinAndLeftRightPosition ? Math.min(visibleItemsCount, seriesCount) : visibleItemsCount) *
+            ITEM_HEIGHT;
+
         return (
             <div className={classNames}>
-                <div className="series" style={{ height: visibleItemsCount * ITEM_HEIGHT }}>
+                <div className="series" style={{ height: seriesHeight }}>
```

Comment:
In this case, we should pass this `seriesHeight` as React prop which its default value is `visibleItemsCount * ITEM_HEIGHT`

Reply:
updated